### PR TITLE
CppCheck 2nd attempt at HWBaseboardRadiator

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes CppCheck issues
 - Supersedes #10019

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/HWBaseboardRadiator.cc [hunk 1]
```diff
@@ -340,27 +303,22 @@ namespace HWBaseboardRadiator {
...
-            VerifyUniqueBaseboardName(
+            GlobalNames::VerifyUniqueBaseboardName(
                 state, cCMO_BBRadiator_Water_Design, state.dataIPShortCut->cAlphaArgs(1), ErrorsFound, cCMO_BBRadiator_Water_Design + " Name");
 
-            HWBaseboardDesignObject(BaseboardDesignNum).designName = state.dataIPShortCut->cAlphaArgs(1); // Name of this baseboard design object
+            thisHWBaseboardDesign.designName = state.dataIPShortCut->cAlphaArgs(1); // Name of this baseboard design object
             HWBaseboardDesignNames(BaseboardDesignNum) = state.dataIPShortCut->cAlphaArgs(1);
 
             // Determine HW radiant baseboard heating design capacity sizing method
             if (UtilityRoutines::SameString(state.dataIPShortCut->cAlphaArgs(iHeatCAPMAlphaNum), "HeatingDesignCapacity")) {
```
Comment (amirroth) [inline src/EnergyPlus/HWBaseboardRadiator.cc]: `enum` + `getEnumerationValue`

File: src/EnergyPlus/HWBaseboardRadiator.cc [hunk 2]
```diff
@@ -423,66 +374,67 @@ namespace HWBaseboardRadiator {
...
-            HWBaseboardDesignObject(BaseboardDesignNum).Offset = state.dataIPShortCut->rNumericArgs(3);
+            thisHWBaseboardDesign.Offset = state.dataIPShortCut->rNumericArgs(3);
             // Set default convergence tolerance
-            if (HWBaseboardDesignObject(BaseboardDesignNum).Offset <= 0.0) {
+            if (thisHWBaseboardDesign.Offset <= 0.0) {
                 ShowWarningError(state,
                                  format("{}{}=\"{}\", {} was less than the allowable minimum.",
                                         RoutineName,
                                         cCMO_BBRadiator_Water_Design,
                                         state.dataIPShortCut->cAlphaArgs(1),
```
Comment (amirroth) [inline src/EnergyPlus/HWBaseboardRadiator.cc]: `thisHWBaseboardDesign.Name`

File: src/EnergyPlus/HWBaseboardRadiator.cc [hunk 3]
```diff
@@ -500,29 +452,26 @@ namespace HWBaseboardRadiator {
...
-            HWBaseboardDesignData HWBaseboardDesignDataObject{
-                HWBaseboardDesignObject(HWBaseboard(BaseboardNum).DesignObjectPtr)}; // Contains the data for the design object
+            thisHWBaseboard.designObjectName = state.dataIPShortCut->cAlphaArgs(2); // Name of the design object for this baseboard
+            thisHWBaseboard.DesignObjectPtr = UtilityRoutines::FindItemInList(thisHWBaseboard.designObjectName, HWBaseboardDesignNames);
+            HWBaseboardDesignData &HWBaseboardDesignDataObject =
+                state.dataHWBaseboardRad->HWBaseboardDesignObject(thisHWBaseboard.DesignObjectPtr); // Contains the data for the design object
 
             // Get schedule
-            HWBaseboard(BaseboardNum).Schedule = state.dataIPShortCut->cAlphaArgs(3);
+            thisHWBaseboard.Schedule = state.dataIPShortCut->cAlphaArgs(3);
```
Comment (amirroth) [inline src/EnergyPlus/HWBaseboardRadiator.cc]: Don't need this if you are immediately getting the schedule index.

File: src/EnergyPlus/HWBaseboardRadiator.cc [hunk 4]
```diff
@@ -664,74 +613,73 @@ namespace HWBaseboardRadiator {
...
+                thisHWBaseboard.TotSurfToDistrib = 0; // error
             }
 
-            HWBaseboard(BaseboardNum).SurfaceName.allocate(HWBaseboard(BaseboardNum).TotSurfToDistrib);
-            HWBaseboard(BaseboardNum).SurfaceName = "";
-            HWBaseboard(BaseboardNum).SurfacePtr.allocate(HWBaseboard(BaseboardNum).TotSurfToDistrib);
-            HWBaseboard(BaseboardNum).SurfacePtr = 0;
-            HWBaseboard(BaseboardNum).FracDistribToSurf.allocate(HWBaseboard(BaseboardNum).TotSurfToDistrib);
-            HWBaseboard(BaseboardNum).FracDistribToSurf = 0.0;
+            thisHWBaseboard.SurfaceName.allocate(thisHWBaseboard.TotSurfToDistrib);
```
Comment (amirroth) [inline src/EnergyPlus/HWBaseboardRadiator.cc]: Do you really need to save the surface names if you are immediately converting them to surface indices below?

File: src/EnergyPlus/HWBaseboardRadiator.cc [hunk 5]
```diff
@@ -664,74 +613,73 @@ namespace HWBaseboardRadiator {
...
-                                                                       HWBaseboard(BaseboardNum).ZonePtr,
-                                                                       HWBaseboard(BaseboardNum).SurfaceName(SurfNum),
-                                                                       ErrorsFound);
-                HWBaseboard(BaseboardNum).FracDistribToSurf(SurfNum) = state.dataIPShortCut->rNumericArgs(SurfNum + 4);
-                if (HWBaseboard(BaseboardNum).FracDistribToSurf(SurfNum) > MaxFraction) {
+            for (SurfNum = 1; SurfNum <= thisHWBaseboard.TotSurfToDistrib; ++SurfNum) {
+                thisHWBaseboard.SurfaceName(SurfNum) = state.dataIPShortCut->cAlphaArgs(SurfNum + 5);
+                thisHWBaseboard.SurfacePtr(SurfNum) = HeatBalanceIntRadExchange::GetRadiantSystemSurface(state,
+                                                                                                         cCMO_BBRadiator_Water,
+                                                                                                         thisHWBaseboard.EquipID,
```
Comment (amirroth) [inline src/EnergyPlus/HWBaseboardRadiator.cc]: Do the world (at least the EnergyPlus development team) a favor and change `EquipID` to `Name`.

File: src/EnergyPlus/HWBaseboardRadiator.cc [hunk 6]
```diff
@@ -893,65 +842,46 @@ namespace HWBaseboardRadiator {
...
             // Initialize the environment and sizing flags
             state.dataHWBaseboardRad->MyEnvrnFlag.dimension(NumHWBaseboards, true);
             state.dataHWBaseboardRad->MySizeFlag.dimension(NumHWBaseboards, true);
-            ZeroSourceSumHATsurf.dimension(state.dataGlobal->NumOfZones, 0.0);
-            QBBRadSource.dimension(NumHWBaseboards, 0.0);
-            QBBRadSrcAvg.dimension(NumHWBaseboards, 0.0);
-            LastQBBRadSrc.dimension(NumHWBaseboards, 0.0);
-            LastSysTimeElapsed.dimension(NumHWBaseboards, 0.0);
-            LastTimeStepSys.dimension(NumHWBaseboards, 0.0);
+            state.dataHWBaseboardRad->ZeroSourceSumHATsurf.dimension(state.dataGlobal->NumOfZones, 0.0);
```
Comment (amirroth) [inline src/EnergyPlus/HWBaseboardRadiator.cc:851]: 👍

File: src/EnergyPlus/HWBaseboardRadiator.cc [hunk 7]
```diff
@@ -1855,26 +1646,26 @@ namespace HWBaseboardRadiator {
...
-                                    DataPlant::CriteriaType::MassFlowRate,
-                                    HWBaseboard(BaseboardNum).WaterMassFlowRate);
-
-        PullCompInterconnectTrigger(state,
-                                    HWBaseboard(BaseboardNum).plantLoc,
-                                    HWBaseboard(BaseboardNum).BBInletTempFlowReSimIndex,
-                                    HWBaseboard(BaseboardNum).plantLoc,
-                                    DataPlant::CriteriaType::Temperature,
-                                    HWBaseboard(BaseboardNum).WaterOutletTemp);
+        PlantUtilities::PullCompInterconnectTrigger(state,
```
Comment (amirroth) [inline src/EnergyPlus/HWBaseboardRadiator.cc:1635]: `auto &hwBaseboard`

File: src/EnergyPlus/HWBaseboardRadiator.cc (hunk 1 above)
Comment (rraustad) [reply src/EnergyPlus/HWBaseboardRadiator.cc]: I at least set up the structure of using enums here.

File: src/EnergyPlus/HWBaseboardRadiator.cc (hunk 6 above)
Comment (rraustad) [reply src/EnergyPlus/HWBaseboardRadiator.cc:851]: I agree but this got reverted out while trying to get back to a clean code base. I had such problems with this branch that after I got back to a clean state I was being overly cautious with any further changes. e.g., the commit names with no diffs and unit tests run means I compiled each change, ran unit tests, and ran the regression tool on the 4 baseboard files. If those tests passed I made the incremental commit. Very time consuming.

**Comment by rraustad** [comment]
Now this branch is back to the same type of diffs as the original branch (3 here and 6 there). Well, of course it is because this branch is at nearly the same state as the other branch. Windows CI does not show these diffs, and I didn't see these diffs locally on any single commit. Go figure. It's one of those last 6 commits. I actually don't think there is anything wrong with these changes, they just cause a slightly different result.

**Comment by mjwitte** [comment]
> Windows CI does not show these diffs

Windows CI does not run any regressions, so it never shows diffs.

**Comment by rraustad** [comment]
I misspoke, I meant "Windows does not show these diffs" because I ran local regressions on 4 baseboard files (5ZoneCoolingPanel*) for **each** of those 6 commits and found 0 diffs each time. 

image

image

**Comment by mjwitte** [comment]
And, obvious question, did you set "force design days" on the local regressions?

**Comment by rraustad** [comment]
Yes. Is that bad?

**Comment by rraustad** [comment]
"Don't force anything" also shows 0 diffs.

**Comment by rraustad** [comment]
"Force Annual simulaitons" also shows 0 diffs.

**Comment by mjwitte** [comment]
> Yes. Is that bad?

No, CI does force design days (except for the few annual simulation that in the regression suite). Sometime I forget to set that locally and end up with different results than CI.

Next question, are you building release exe's for the local regressions? If release exists, the regression tool will use it even if it's not current.

**Comment by rraustad** [comment]
I built release, ran local units tests, and ran the regression tool, for each of those commits with name ending in "no diffs and unit tests run".

**Comment by rraustad** [comment]
Something is wrong with that workflow. I did a clean and build and now I see diffs.

**Comment by Myoldmopar** [comment]
Pulled develop in locally and tested it, everything looks good.  And everything was good the last time Decent ran here.  This is good to go in.  Thanks for the extra effort here with the branch restart @rraustad.

**Pull request merged**